utils/check_code_style.py

```diff
@@ -0,0 +1,587 @@
+		["Foundation, either version 3 of the License, or (at your option) any later version.", False],
+		["", False],
+		["Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY", False],
+		["WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A", False],
+		["PARTICULAR PURPOSE. See the GNU General Public License for more details.", False],
+		["", False],
+		["You should have received a copy of the GNU General Public License along with", False],
+		["this program. If not, see <https://www.gnu.org/licenses/>.", False],
+		["*/", False],
+		["", False]
+	]
+	index = 0
+	error_line = -1
+	complete = False
+	for [copyright, is_regex] in copyright_begin:
+		if is_regex:
+			if not re.search(copyright, lines[index]):
+				error_line = index
+				break
+		else:
+			if copyright != lines[index]:
+				error_line = index
+				break
+		index += 1
+	not_found_error_line = index
+	if error_line == -1:
+		index_begin = index
+		while index_begin < len(lines) - len(copyright_end):
+			index = index_begin
+			for [copyright, is_regex] in copyright_end:
+				if is_regex:
+					if not re.search(copyright, lines[index]):
+						error_line = index
+						break
+				else:
+					if copyright != lines[index]:
+						error_line = index
+						break
+				index += 1
+			if error_line == -1:
+				complete = True
+				break
+			index_begin += 1
+			error_line = -1
+	if error_line != -1:
+		errors.append(Error(lines[error_line], error_line + 1, "invalid or missing copyright header"))
+	elif not complete:
+		errors.append(Error(lines[not_found_error_line], not_found_error_line + 1,
+							"invalid or incomplete copyright header"))
+	return errors, warnings
+
+
+# Checks whether the specified lines don't exceed the character limit, and consist of ASCII characters only. Parameters:
+# lines: the lines to check, without the terminating line separators
+# Returns a tuple of errors and warnings.
+def check_line_format(lines):
+	errors = []
+	warnings = []
+
+	line_count = 0
+	for line in lines:
+		line_count += 1
+		if len(line) > 120:
+			errors.append(Error(line, line_count, "lines should hard wrap at 120 characters"))
+		for char in line:
+			if ord(char) < 0 or ord(char) > 127:
+				errors.append(Error(line, line_count, "files should be plain ASCII"))
+				break
+	return errors, warnings
+
+
+# Checks the import statements at the beginning of the file. Parameters:
+# sanitized_lines: the lines of the file, without the line separators and the contents of strings and comments
+# original_lines: the lines of the file, without the terminating line separators
+# file: the path to the file
+# Returns a tuple of errors and warnings.
+def check_include(sanitized_lines, original_lines, file):
+	errors = []
+	warnings = []
+
+	# opengl.h is treated as a <> include in most cases
+	original_lines = [line if line != "#include \"opengl.h\"" else "#include <opengl.h>" for line in original_lines]
```

Comment:
Please add this to a separate "ignore list". The style-checking code and the list of exceptions should be separate files.

Reply:
I've added a new list at the beginning of the file which changes the type of the include statement (currently only `opengl.h`), so it is now configurable. I've also added a new list that skips include checks for an entire file (currently, that is only `main.cpp`). I doubt any of these lists would see a lot of changes in the future so I didn't separate them into different files, but it should now be pretty straightforward to exclude new ones when needed.

Reply:
Ok, let's see if this is sufficient. And if not, then we need a follow-up PR where we have more options for making exceptions.